code/game/objects/items/reagent_containers/hypospray.dm

```diff
@@ -16,72 +16,286 @@
 	flags_equip_slot = SLOT_WAIST
 	w_class = 2.0
 	var/skilllock = 1
+	var/inject_mode = TRUE //TRUE for inject FALSE for drain
+	var/core_name = "hypospray"
+	var/label = null
+
+/obj/item/reagent_container/hypospray/advanced/attack_self(mob/user)
+	if(user.is_mob_incapacitated() || !usr.IsAdvancedToolUser())
+		return FALSE
+
+	label(user)
+
+/obj/item/reagent_container/hypospray/proc/empty(mob/user)
+	if (alert(user, "Are you sure you want to empty [src]?", "Flush [src]:", "Yes", "No") != "Yes")
+		return
+	if(isturf(user.loc))
+		to_chat(user, "<span class='notice'>You flush the contents of [src].</span>")
+		reagents.reaction(user.loc)
+		reagents.clear_reagents()
+
+/obj/item/reagent_container/hypospray/proc/label(mob/user)
+	var/str = copytext(reject_bad_text(input(user,"Label text?", "Set label", "")), 1, MAX_NAME_LEN)
+	if(!str || !length(str))
+		to_chat(user, "<span class='notice'>Invalid text.</span>")
+		return
+	to_chat(user, "<span class='notice'>You label [src] as \"[str]\".</span>")
+	name = "[core_name] ([str])"
+	label = str
 
 /obj/item/reagent_container/hypospray/attack_paw(mob/user as mob)
 	return attack_hand(user)
 
-/obj/item/reagent_container/hypospray/attack(mob/M, mob/living/user)
+/obj/item/reagent_container/hypospray/afterattack(obj/T, mob/living/user)
+	if(!T.reagents)
+		return
+	if(!inject_mode) //if we're draining
+		if(reagents.holder_full())
+			to_chat(user, "<span class='warning'>[src] is full.</span>")
+			return
+
+		if(ismob(T))//Blood!
```

Comment:
just remove the ismob check and do iscarbon since you're not doing anything different if they aren't carbon.